### ISSUE TITLE
RAM not relesase

### ISSUE THREAD

**Issue by Hank0223** [AlexeyAB/darknet#4376]
I tried to train in darknet classifier, but I found that RAM can't be released, which eventually caused the entire computer to crash.

My computer equipment 
Graphics card RTX2080
CPU i9 9900K

Environmental configuration
NVIDIA 440.36
CUDA 10.2
CUDNN 7.6

**Comment by AlexeyAB** [comment]
Do you use the latest Darknet version?
Do you use Windows or Linux?

**Comment by Hank0223** [comment]
Yes, I use the latest Darknet version.
I execute on Linux.

Ubuntu 16.04 and Ubuntu18.04 all happen the same thing.

**Comment by AlexeyAB** [comment]
Can you show screenshot when Darknet is crashed?
What params do you use in Makefile?
Do you use only 1 GPU?
What CUDA, cuDNN and OpenCV versions do you use?

**Comment by Hank0223** [comment]
After the darknet crashes, the computer will not respond to the keyboard and mouse, so it will not be able to show screenshot.

Yes, I use only 1 GPU
CUDA 10.2.89, cuDNN 7.6.5, OpenCV3.4

[test.zip](https://github.com/AlexeyAB/darknet/files/3886436/test.zip)

**Comment by Hank0223** [comment]
I use GPU=1, cuDNN=1 ,OpenCV=1 in the Makefile.

cfg file
[test.zip](https://github.com/AlexeyAB/darknet/files/3886457/test.zip)

**Comment by AlexeyAB** [comment]
There functions are called only once for the first detection: https://github.com/AlexeyAB/darknet/blob/dd4dcf20b4886f4cfedb82e6d205a93c3a4c1034/src/gemm.c#L684-L704

**Comment by AlexeyAB** [comment]
Do you get this issue if you remove `xnor=1` in all [conv] layers?

**Comment by Hank0223** [comment]
Yes, I still encountering this issue

**Comment by AlexeyAB** [comment]
Try to download new version of Darknet.

**Comment by Hank0223** [comment]
The new version of Darknet, still encountering the same problem
![Screenshot from 2019-11-25 21-16-30](https://user-images.githubusercontent.com/58155722/69543760-42b19980-0fc9-11ea-9028-faee3ed93b28.png)

**Comment by Hank0223** [comment]
I found that I will solve this problem after OpenCV=0, maybe the problem is in OpenCV.

**Comment by AlexeyAB** [comment]
There was a bug in Classifier data augmentation, but it was solved.
Did you do `make clean` before `make` after loading the new Darknet?


Can you install Valgrind (for checking memory leaks)
```
sudo apt update
sudo apt install valgrind
```
then compile Darknet with OPENCV=1, 
set `max_batches = 100` in cfg-file 
and run training with valgrind?

`valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes --verbose --log-file=valgrind-out.txt ./darknet classifier train ....`

then attach `valgrind-out.txt` file

**Comment by Hank0223** [comment]
valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes --verbose --log-file=valgrind-out.txt ./darknet classifier train ....
I use this command, training doesn't work.
Training becomes very slow.GPU doesn't work properly

**Comment by AlexeyAB** [comment]
So `it doesn't work` or `it works very slow`?

Set max_batches=10  - it should take much time, and it will stop automatically after 10 iterations.

**Comment by Hank0223** [comment]
Sorry, it working very slow and crashes later.
[valgrind-out.txt](https://github.com/AlexeyAB/darknet/files/3889253/valgrind-out.txt)

**Comment by Hank0223** [comment]
Is there a problem with the version configuration?

**Comment by tigerdhl** [comment]
@AlexeyAB  i have the same issue,it stops at radomly iteration when training,and my computer is crash. I use the yolov4-tiny, I clone the project 3 day ago, i think is a new version, my device:
GTX 2080Ti
GPU=1
CUDNN=1
OPENCV=1
![Dingtalk_20201209111316](https://user-images.githubusercontent.com/29905731/101571502-bc6cc780-3a12-11eb-854b-4b2ce8602f70.jpg)

[net]
# Testing
#batch=1
#subdivisions=1
# Training
batch=64
subdivisions=1
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.00261
burn_in=1000
max_batches = 14000
policy=steps
steps=11200,12600
scales=.1,.1

[convolutional]
batch_normalize=1
filters=32
size=3
stride=2
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=64
size=3
stride=2
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[route]
layers=-1
groups=2
group_id=1

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[route]
layers = -1,-2

[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=1
activation=leaky

[route]
layers = -6,-1

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[route]
layers=-1
groups=2
group_id=1

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[route]
layers = -1,-2

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[route]
layers = -6,-1

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[route]
layers=-1
groups=2
group_id=1

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[route]
layers = -1,-2

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[route]
layers = -6,-1

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

##################################

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=36
activation=linear



[yolo]
mask = 3,4,5
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=7
num=6
jitter=.3
scale_x_y = 1.05
cls_normalizer=1.0
iou_normalizer=0.07
iou_loss=ciou
ignore_thresh = .7
truth_thresh = 1
random=0
resize=1.5
nms_kind=greedynms
beta_nms=0.6

[route]
layers = -4

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[upsample]
stride=2

[route]
layers = -1, 23

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=36
activation=linear

[yolo]
mask = 0,1,2
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=7
num=6
jitter=.3
scale_x_y = 1.05
cls_normalizer=1.0
iou_normalizer=0.07
iou_loss=ciou
ignore_thresh = .7
truth_thresh = 1
random=0
resize=1.5
nms_kind=greedynms
beta_nms=0.6


I set the OPENCV=0,it is also occured, it seems like the memory leak. How to fix it?
